shared_model/validators/field_validator.hpp

```diff
@@ -160,33 +203,46 @@ namespace shared_model {
         iroha::ts64_t now = iroha::time::now();
         // TODO 06/08/17 Muratov: make future gap for passing timestamp, like
         // with old timestamps IR-511 #goodfirstissue
+
         if (now < timestamp) {
-          reason.second.push_back(boost::str(
-              boost::format(
-                  "timestamp broken: send from future (%llu, now %llu)")
-              % timestamp % now));
+          auto message = (boost::format("bad timestamp: sent from future, "
+                                        "timestamp: %llu, now: %llu")
+                          % timestamp % now)
+                             .str();
+          reason.second.push_back(std::move(message));
```

Comment:
this branch is not covered by tests

Reply:
Right now it is hard to test timestamps, because we cannot verify that message is correct. I will think of a solution and maybe do a separate pull request